src/hotspot/os/bsd/os_perf_bsd.cpp

```diff
@@ -316,6 +316,34 @@ int CPUPerformanceInterface::CPUPerformance::context_switch_rate(double* rate) {
   _jvm_context_switches = jvm_context_switches;
   _total_csr_nanos = total_csr_nanos;
 
+  return result;
+#elif defined(__FreeBSD__)
+  long jvm_context_switches;
```

Comment:
This doesn't work correctly.  I've got a local change to do this and using a long results in incorrect values for the context switches.  From sysctl(3):

"The information available from sysctl() consists of integers, strings, and tables."

Using a long only works for integer data if a long and int are the same size.  Otherwise you get garbage in the other bits.